src/sortStudents.ts

```diff
@@ -1,16 +1,52 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
+  averageGrade: number;
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudens = [...students];
+
+  sortedStudens.forEach((student) => {
+    const changedStudent = student;
+
+    changedStudent.averageGrade = student.grades
+      .reduce((sum, grade) => sum + grade) / student.grades.length;
+  });
+
+  sortedStudens.sort((firstStudent: Student, secondStudent: Student) => {
+    switch (order) {
+      case 'asc':
+        return firstStudent[sortBy] < secondStudent[sortBy]
+          ? -1
+          : 1;
+
+      case 'desc':
+        return firstStudent[sortBy] > secondStudent[sortBy]
+          ? -1
+          : 1;
```

Comment:
This point in task mean that you should build your code using kind of this logic:

<img width="419" alt="Screenshot 2022-09-02 at 10 30 32" src="https://user-images.githubusercontent.com/17098736/188082968-36a84976-57f4-4179-8535-8d8038d6dfef.png">

```suggestion
    switch (sortBy) {
    case SortType.Name:
    case SortType.Surname:
      return order === 'asc'
        ? ...
        : ...

    case SortType.Age:
    case SortType.Married:
      return order === 'asc'
        ? ...
        : ...

```